operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java

```diff
@@ -66,7 +66,7 @@ protected ReconcileResult<R> reconcile(P primary, R actualResource, Context<P> c
         if (updatable) {
           final Matcher.Result<R> match = match(actualResource, primary, context);
           if (!match.matched()) {
-            final var desired = match.computedDesired().orElse(desired(primary, context));
+            final var desired = match.computedDesired().orElseGet(() -> desired(primary, context));
```

Comment:
This is the salient part of the fix. I also took the opportunity to fix similar issues and low hanging fruits.